pypesto/objective/amici_objective.py

```diff
@@ -84,17 +79,31 @@ def sres(x):
         # extract parameter names from model
         self.x_names = list(self.amici_model.getParameterNames())
 
+    def fun(self, x):
+        return self.call_amici(
+            x,
+            Objective.MODE_FUN
+        )
+
+    def res(self, x):
```

Comment:
is this overwriting of fun, res necessary? codacy complains and it might be error-prone and is also not so nice to have ``overwritefun`` in the constructor. i think it is valid to save the sensi_orders temporarily, as you did in the Objective. But maybe one can just have in ``__init__``
```
def fun(x):
    return self._call_amici(x, Objective.MODE_FUN)
...
super().__init__(...fun=fun,...)
```
and similarly for res?